src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -166,12 +184,19 @@ trait IterableMonoTransforms[+A, +Repr] extends Any {
   def drop(n: Int): Repr = fromIterableWithSameElemType(View.Drop(coll, n))
 
   /** The rest of the collection without its first element. */
-  def tail: Repr = drop(1)
+  def tail: Repr = if (coll.isEmpty) ??? else drop(1)
```

Comment:
@szeiger I’m not eliminating `tail`, I’m fixing its implementation. It is not supposed to terminate when called on an empty list.

Concerning the non-emptiness, we also have `+:` and `:+`, and we could overload the `apply` factory method to return a non-empty collection if called with at least one element.